### DISCUSSION TITLE
Terminal is not cleaned up when I quit the fastAPI server with ctrl+c

### DISCUSSION THREAD

**Discussion by jsrimr** [fastapi/fastapi#7302] [Questions] [answered]
![image](https://user-images.githubusercontent.com/16870927/93410354-cc718100-f8d3-11ea-9ede-b421e765e7dc.png)

Like above picture, I finished the server with ctrl+c. However, the terminal I was using is still freezed, so I need to open another fresh terminal if I want to start the server again.

Temporarily, I find the fastAPI process with ```ps -ef | grep fast``` and kill that process with ```kill -9 [pid]```. This ends up the server and leave terminal cleanly, but too burdensome. 

Is there anyway to use ctrl+c more smartly or any other practical alternatives? Thx

**Comment by raphaelauv** [comment]
Do you have a working example ?

Ctrl+c is working , take a look there : 
https://github.com/tiangolo/fastapi/issues/2025#issuecomment-689682885

**Comment by jsrimr** [comment]
I tried to provide simplified version of my code which contains some thread jobs and DB I/O tasks but it works fine with Ctrl+c ...

Still, I cannot gracefully end the original fastAPI process with Ctrl+c. Could you provide any guessing ideas for me?

**Comment by raphaelauv** [comment]
No , at least try the example I gave you , maybe it's your terminal configuration.

**Comment by jsrimr** [comment]
Sorry for the late reply. I tried your code and it terminated gracefully. It doesn't seem a matter of terminal configuration. I will research more and come next time.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the help here @raphaelauv! 🍪 

And thanks for coming back to close the issue @jsrimr ☕ 

...there's a chance that you are on Windows and you had selected text in the terminal. In Windows, that freezes everything until you hit enter, that copies that text and lets things continue working... because Windows 😅 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
